### PR TITLE
Add the migraphx integration script as a jenkins pipeline

### PR DESCRIPTION
**Pull request by giuseros**
Fixes https://github.com/ROCmSoftwarePlatform/rocMLIR-internal/issues/1172

### CODE REVIEW

**Comment by giuseros** [comment]
This pipeline has been tested here: http://rocmhead.amd.com:8080/job/MLIR/job/test/18/

File: mlir/utils/jenkins/Jenkinsfile.migraphxintegration [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+    }
+    post {
+        success {
+            mail to: "giuseppe.rossini@amd.com,manupa.karunaratne@amd.com,krzysztof.drewniak@amd.com",
+             subject: "MIGraphX integration job passed!",
+             body: "Done"
+        }
+        failure {
+            mail to: "giuseppe.rossini@amd.com,manupa.karunaratne@amd.com,krzysztof.drewniak@amd.com",
+             subject: "Ouch! MIGraphX integration job failed!",
```
Comment (krzysz00) [inline mlir/utils/jenkins/Jenkinsfile.migraphxintegration:227]: Could you refactor this into a Jenkins pipeline where you have Jenkins spawn the Docker container?

Comment (giuseros) [reply mlir/utils/jenkins/Jenkinsfile.migraphxintegration:227]: I was thinking to do that after this was merged. The idea of the ticket was to bring the script I wrote to test MIGraphX into our repository. I can refactor it at a later stage (please note that this won't even run in any nightly job for now)

Comment (krzysz00) [reply mlir/utils/jenkins/Jenkinsfile.migraphxintegration:227]: Fair enough

**Review by krzysz00 — APPROVED** [review]

**Pull request merged**